### ISSUE TITLE
Add option to disable browserify build

### ISSUE THREAD

**Issue by MoOx** [defunctzombie/zuul#158]
I already have a builded test file (I am using webpack, mainly to handle all assets & have react hot reload). So I would like to be able to just run this test file with zuul. But it seems (it's not clear in the README nor the doc) that the files passed are automatically "browserified".

Would you be open to add an simple option to skip the building part ?

**Comment by vvo** [comment]
Did you try:

test.js
```js
require('./your-bunlded-file.js');
```

?

**Comment by MoOx** [comment]
I just yes. But I think browserify is failling silently cause this is falling : `browserify test.js`

**Comment by MoOx** [comment]
zuul does not output any error :/

**Comment by vvo** [comment]
Are you trying to use phantomjs? Phantomjs lacks global error handle I think.

**Comment by vvo** [comment]
Can you try to `browserify test.js` and see why bundling your test.js would fail?

**Comment by MoOx** [comment]
I'm using `zuul --local --open -- test.js`, the browser open, but output nothing. When I view the source I don't see any reference to a bundled file (but maybe I should not see any).
Indeed browserify is failling, on webpack loader syntax. I will see what I can do by myself. Thanks for helping.

**Comment by vvo** [comment]
Another solution would be to provide an empty test.js and then use the `scripts` option in zuul.yml https://github.com/defunctzombie/zuul/wiki/Zuul.yml#scripts-optional

Problem is, `scripts` was designed to be "required scripts to launch the tests". So they are put before the zuul testing client. Thus your (tape?) tests would begin before the zuul client is launched.

We might have an option like before_scripts, after_scripts and it would solve your issue.

But really you are in a corner case where you try to use zuul in a way it was not written for.

**Comment by vvo** [comment]
One solution that would work would be to have:

test.js
```js
var load = require('load-script')

load('super-bundle.js', function (err) {
  if (err) {
    // print useful message
  }
  else {
    // use script
    // note that in IE8 and below loading error wouldn't be reported
  }
})
```

This should work but way hacky.

We are using watchify in the browserify builder, so you won't benefit from it with this strategy and you might get caching issues.

Last, the buidler is a zuul.yml option, see https://github.com/defunctzombie/zuul/blob/0ed6ff51932d29a24d07332d7a2f6d4901199935/lib/control-app.js#L44, the interface is: https://github.com/defunctzombie/zuul/blob/0ed6ff51932d29a24d07332d7a2f6d4901199935/lib/control-app.js#L125

So maybe you can write a webpack builder :)

**Comment by MoOx** [comment]
We should just be able to ask zuul to handle raw scripts & watch them for changes :)
webpack-dev-server already handle a lot of things (server, caching, fast rebuild etc)
But yeah, I will see what I can do.

**Comment by vvo** [comment]
there is room for improvment on the zuul.yml and zuul API I agree. Zuul code is pretty straightforward on this. ;)

**Comment by MoOx** [comment]
So my code was bloated with some unresolved require() due to how webpack build some stuff with hot reload. With --production flag on, I can make zuul works for me.

That being said, I think it would  be great to add an option to just run the file as it is. So I let this issue open if someone want to handle that optimization ^^

**Comment by defunctzombie** [comment]
@MoOx If you use the "script" feature of zuul.yml (https://github.com/defunctzombie/zuul/wiki/Zuul.yml#scripts-optional) it will do what you want.

We could also look into adding support for a webpack builder or at least reviewing a PR for it if you want to do it :)

**Comment by defunctzombie** [comment]
Alternatively, we could do something with disabling browserify on the test files and just loading raw files (but I think script can do that for you already)

**Comment by MoOx** [comment]
No the script feature execute my tape tests too early. I get 0 test recognized. For now I am just fine with a standalone file that execute tape tests. browserify is used by my CI system and just do nothing since there is no `require()` left.
Locally I use testem (ci mode) which is a great launcher for local browsers in precommit hook (cover less browsers, but is way more faster, tests take just some seconds instead of 30min for all tests on my browsers scope on saucelab :) )

**Comment by rase-** [comment]
@defunctzombie we may need to enable users to place scripts also after the zuul suite.

**Comment by MoOx** [comment]
Yeah, that should be quick to implement, and fast for end user. Maybe with a simpler watcher on these files (like testem does).

**Comment by vvo** [comment]
> @defunctzombie we may need to enable users to place scripts also after the zuul suite.

Yes, tape tests will not work when placed before client.js
